### PR TITLE
Rasterize Vector Data

### PR DESCRIPTION
**Pull request by GregoryPetrochenkov-NOAA**
This PR worked on the following issue:
[Operations to Rasterize Vector Datasets #94](https://github.com/NOAA-OWP/gval/issues/94)

All tests pass.

## Additions

-  rasterize.py

## Removals

-

## Changes

- Tutorial.ipynb (Merged main)
- requirements/base.txt: Added geocube to base.txt
- gval_xarray.py: Add ability to pass in GeoDataFrame for benchmark map
- cases_spatial_alignment.py: Added cases for testing rasterize vectors
- test_spatial_alignment.py: Added cases for testing rasterize vectors
- conftest.py: Add load geopackage

### CODE REVIEW

File: src/gval/accessors/gval_xarray.py [hunk 1]
```diff
@@ -173,18 +185,27 @@ def spatial_alignment(
...
+        rasterize_attributes: Optional[list], default = None
+            Attributes to rasterize from vector dataset
+
 
         Returns
         --------
         Union[xr.Dataset, xr.DataArray]
             Tuple with candidate and benchmark map respectively.
         """
+        benchmark_map = _rasterize_data(
```
Comment (fernando-aristizabal) [inline src/gval/accessors/gval_xarray.py:203]: Shouldn't this be its own accessor method? I'm not sure if it's within the scope of spatial alignment as now spatial alignment has been given the task of data conversion as well? The other option is we rename this current accessor method to something like homogenize to better reflect the different operations that could take place?

File: tests/test_spatial_alignment.py [hunk 2]
```diff
@@ -136,3 +137,37 @@ def test_spatial_alignment_fail(candidate_map, benchmark_map, target_map, kwargs
...
+def test_rasterize_vector_success(candidate_map, benchmark_map, rasterize_attributes):
+    """Test rasterize vector success"""
+
+    benchmark_raster = _rasterize_data(
+        candidate_map=candidate_map,
+        benchmark_map=benchmark_map,
+        rasterize_attributes=rasterize_attributes,
+    )
+
+    if isinstance(benchmark_raster, xr.Dataset):
```
Comment (fernando-aristizabal) [inline tests/test_spatial_alignment.py:155]: Any tests for data quality?

File: src/gval/homogenize/rasterize.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    """
+
+    if isinstance(benchmark_map, gpd.GeoDataFrame):
+        rasterized_data = make_geocube(
+            vector_data=benchmark_map,
+            measurements=rasterize_attributes,
+            like=candidate_map,
+        )
+
+        if len(rasterized_data.data_vars) < 1:
```
Comment (fernando-aristizabal) [inline src/gval/homogenize/rasterize.py:47]: This checks the variables to ensure they are numeric? Can we do this prior to geocubing?

File: src/gval/accessors/gval_xarray.py [hunk 4]
```diff
@@ -103,12 +106,20 @@ def categorical_compare(
...
             Categories to represent negative entries
+        rasterize_attributes: Optional[list], default = None
+            Attributes to rasterize from vector dataset
 
         Returns
         -------
         Union[xr.Dataset, xr.DataArray], DataFrame[Crosstab_df], DataFrame[Metrics_df]
             Tuple with agreement map, cross-tabulation table, and metric table
         """
+        benchmark_map = _rasterize_data(
```
[the remark this answers is not in the document]
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/accessors/gval_xarray.py]: Refactored out of function.

File: src/gval/accessors/gval_xarray.py (hunk 1 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/accessors/gval_xarray.py]: Changed to be dataframe accessor method.

File: src/gval/homogenize/rasterize.py (hunk 3 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/homogenize/rasterize.py]: Checks are now prior to geocube

File: tests/test_spatial_alignment.py (hunk 2 above)
Comment (GregoryPetrochenkov-NOAA) [reply tests/test_spatial_alignment.py:157]: There are now data quality tests.

**Pull request merged**